src/global/signal_handler.cc

```diff
@@ -166,6 +195,19 @@ void install_standard_sighandlers(void)
  *     are sent twice in quick succession.
  */
 struct SignalHandler : public Thread {
+  
+  safe_handler(bool oneshot=false) {
+    one_shot = oneshot;
+    memset(pipefd, 0, sizeof(pipefd));
+    memset(&handler, 0, sizeof(handler));
+    memset(&siginfo_handler, 0, sizeof(siginfo_handler));
+    memset(&info_t, 0, sizeof(info_t));    
+  }
+
+  signal_siginfo_handler_t siginfo_handler;
+  siginfo_t info_t;
+  bool one_shot;
```

Comment:
I don't understand what one_shot is for. Why not just *only* register the fancy siginfo signal handler?